py/picca/delta_extraction/expected_fluxes/true_continuum.py

```diff
@@ -198,7 +188,99 @@ def compute_expected_flux(self, forests, out_dir):
+        self.save_delta_attributes()
+
+    def compute_mean_cont_lin(self, forests):
+        """Compute the mean quasar continuum over the whole sample assuming a
+        linear wavelength solution. Then updates the value of self.get_mean_cont
+        to contain it
+
+        Arguments
+        ---------
+        forests: List of Forest
+        A list of Forest from which to compute the deltas.
+        """
+        num_bins = self.lambda_rest_frame.size
+        mean_cont = np.zeros(num_bins)
+        mean_cont_weight = np.zeros(num_bins)
+
+        for forest in forests:
+            if forest.bad_continuum_reason is not None:
+                continue
+            bins = (
+                (forest.lambda_ /
+                 (1 + forest.z) - Forest.lambda_min_rest_frame) /
+                (Forest.lambda_max_rest_frame - Forest.lambda_min_rest_frame) *
+                num_bins).astype(int)
+
+            var_lss = self.get_var_lss(forest.lambda_)
+            var_pipe = 1. / forest.ivar / forest.continuum**2
+            variance = var_lss + var_pipe
+            weights = 1 / variance
+            cont = np.bincount(bins,
+                               weights=forest.continuum * weights)
+            mean_cont[:len(cont)] += cont
+
+            cont = np.bincount(bins, weights=weights)
+            mean_cont_weight[:len(cont)] += cont
+
+        w = mean_cont_weight > 0
+        mean_cont[w] /= mean_cont_weight[w]
+        mean_cont /= mean_cont.mean()
+        lambda_cont = self.lambda_rest_frame[w]
+
+        self.get_mean_cont = interp1d(lambda_cont,
+                                      mean_cont,
+                                      fill_value="extrapolate")
+        self.get_mean_cont_weight = interp1d(lambda_cont,
+                                             mean_cont_weight,
+                                             fill_value=0.0,
+                                             bounds_error=False)
+
+    def compute_mean_cont_log(self, forests):
+        """Compute the mean quasar continuum over the whole sample assuming a
+        log-linear wavelength solution. Then updates the value of
+        self.get_mean_cont to contain it
+
+        Arguments
+        ---------
+        forests: List of Forest
+        A list of Forest from which to compute the deltas.
+        """
+        num_bins = self.log_lambda_rest_frame.size
+        mean_cont = np.zeros(num_bins)
+        mean_cont_weight = np.zeros(num_bins)
+
+        for forest in forests:
+            if forest.bad_continuum_reason is not None:
+                continue
+            bins = ((forest.log_lambda - Forest.log_lambda_min_rest_frame -
+                     np.log10(1 + forest.z)) /
+                    (Forest.log_lambda_max_rest_frame -
+                     Forest.log_lambda_min_rest_frame) * num_bins).astype(int)
+
+            var_lss = self.get_var_lss(forest.log_lambda)
+            var_pipe = 1. / forest.ivar/ forest.continuum**2
```

Comment:
is the "continuum" here the product of continuum times "mean transmitted flux fraction"? Otherwise it looks like the next equation is not right, one needs to take into account the "mean transmitted flux fraction" somewhere.

Reply:
Ignasi confirmed that this is "continuum * mean transmitted flux fraction". We should rename some of the variables in picca, but let's leave for a future issue.